File: Pilas/Con_Arreglos/pila_lista.py
MAX_SIZE = 100
stack = [0] * MAX_SIZE
top = -1

def push(item):
    global top
    if top == MAX_SIZE - 1:
        print("Stack Overflow")
        return
    top += 1
    stack[top] = item

def pop():
    global top
    if top == -1:
        print("Stack Underflow")
        return -1
    value = stack[top]
    top -= 1
    return value

def peek():
    if top == -1:
        print("Pila vacía")
        return -1
    return stack[top]

def is_empty():
    return top == -1

def is_full():
    return top == MAX_SIZE - 1

push(10)
push(20)
push(30)
print("Elemento Superior:", peek())
print("Extrae elemento:", pop())
print("Elemento Superior:", peek())

File: Pilas/Con_Arreglos/pila_lista.cs
using System;

class StackArray {
    const int MAX_SIZE = 100;
    static int[] stack = new int[MAX_SIZE];
    static int top = -1;

    static void Push(int item) {
        if (top == MAX_SIZE - 1) {
            Console.WriteLine("Stack Overflow");
            return;
        }
        stack[++top] = item;
    }

    static int Pop() {
        if (top == -1) {
            Console.WriteLine("Stack Underflow");
            return -1;
        }
        return stack[top--];
    }

    static int Peek() {
        if (top == -1) {
            Console.WriteLine("Pila vacía");
            return -1;
        }
        return stack[top];
    }

    static bool IsEmpty() => top == -1;
    static bool IsFull() => top == MAX_SIZE - 1;

    static void Main() {
        Push(10);
        Push(20);
        Push(30);

        Console.WriteLine("Elemento Superior: " + Peek());
        Console.WriteLine("Extrae elemento: " + Pop());
        Console.WriteLine("Elemento Superior: " + Peek());
    }
}